devito/propagator.py

```diff
@@ -83,11 +85,14 @@ def __init__(self, name, nt, shape, spc_border=0, time_order=0,
 
         # Cache blocking and default block sizes
         self.cache_blocking = cache_blocking
-        if(isinstance(block_size, Iterable)):
-            if(len(block_size) == len(shape)):
+        self.cache_block_inner = cb_inner_dim
+        if isinstance(block_size, Iterable):
+            if len(block_size) == len(shape):
                 self.block_sizes = block_size
+            elif len(block_size) == len(shape) - 1 and not self.cache_block_inner:
```

Comment:
Yes, it is neccessary because with current implementation we can pass single `block_size`, which would give square blocks on all dimensions. Thus the flag is required, if we do not want to block on all dimensions in that case.
I added this statement here to append dummy value because with current implementation `block_size` is zipped with `shape `and then iterated over to create blocking loops. Thus, length of them has to be the same when creating loops.